### DISCUSSION TITLE
How do i use DB::transactions inside the form ?

### DISCUSSION THREAD

**Discussion by khanakia** [filamentphp/filament#3416] [Help] [answered]
I have a form and when that form is submitted i create some entry in some other table also. So i want to use Laravel DB transactions https://laravel.com/docs/9.x/database#database-transactions if the transaction fails roll back both entries the one created via FORM.

How do i achieve this ?

**Comment by danharrin** [comment, marked as answer]
https://filamentphp.com/docs/2.x/admin/resources/creating-records#lifecycle-hooks
https://laravel.com/docs/9.x/database#manually-using-transactions

**Reply by khanakia** [reply]
Just for reference posting here. I did this way.
```
...

use Illuminate\Support\Facades\DB;

class CreateUserPaymentRequest extends CreateRecord
{
    protected static string $resource = UserPaymentRequestResource::class;

    public function create(bool $another = false): void
    {
        DB::transaction(function () use ($another) {
            parent::create($another);
           
            $data = $this->form->getState();
            Ledger::createEntry([
                'user_id' => auth()->user()->id,
                'amount' => -1* $data['amount'],
            ]);
        });
    }
}

```
